### ISSUE TITLE
Can't boot kernel built using build-kernel-qemu

### ISSUE THREAD

**Issue by hpeyerl** [dhruvvyas90/qemu-rpi-kernel#7]
Using a raspbian-ua-netinst-v1.0.8.1.img, I can boot using both kernel-qemu-4.1.7-jessie and kernel-qemu-4.1.13-jessie which is awesome.  Unfortunately, after install is finished, systemd has trouble with dev-sda1.device and dev-ttyAMA0.device; presumably due to 'CONFIG_FHANDLE is not set'.  So I modified build-kernel-qemu to include 'CONFIG_FHANDLE=y' and rebuilt the kernel.  Unfortunately, on boot I get:

Error: unrecognized/unsupported processor variant (0x410fb767).

I get the same error if I try to build without modifications.  

Is this the script that was actually used to build kernel-qemu-4.1.7-jessie and qemu-4.1.13-jessie?   If so, does anyone has a .config I can borrow to compare against?

**Comment by dhruvvyas90** [comment]
May I have the link of the image file that you're using ?

**Comment by hpeyerl** [comment]
https://github.com/debian-pi/raspbian-ua-netinst/releases/tag/v1.0.8.1

It needs to be prepared using all the usual s/mmcblk0p1/sda1/ and ld.so.preload stuff...  Also add/grow a second partition to about 500MB's.   I can post the prep script I used after I get home.

That's an installer image.  After you boot it, it will apt-get install a bunch of stuff and then try to reboot to multi-user... Reboot with root=/dev/sda2 and

**Comment by hpeyerl** [comment]
Here's the prep script I used:

https://github.com/JedMeister/playing-with-rpi/blob/master/raspbian-ua-netinst-qemu-prep.sh

**Comment by hpeyerl** [comment]
Here is the problematic output when booting one of your pre-built kernels.. 

[ ***  ] (1 of 2) A start job is running for dev-sda1.devi...41s / 1min 30s)
[ ***  ] (2 of 2) A start job is running for dev-ttyAMA0.devi...43s / 1min 30s)

**Comment by dhruvvyas90** [comment]
@hpeyerl: Were you able to find a fix for your particular rootfs ?

**Comment by hpeyerl** [comment]
No.  I would have to downgrade systemd to something from 2012 and I wasn't able to find a binary package in the repo for it.

**Comment by rimmington** [comment]
@hpeyerl In case you're still working on this:

* I ran into the same issue re ttyAMA0. My branch with fixes is at https://github.com/rimmington/qemu-rpi-kernel/tree/4.1.18-fhandle
* I've uploaded a `.config` that works for me with Raspbian Jessie lite at https://gist.github.com/rimmington/a72c83348b9d5859c62e83aa03540f03
* For reference, the `qemu` command I'm using: `qemu-system-arm -kernel ./kernel-qemu -cpu arm1176 -m 256 -M versatilepb -append "root=/dev/sda2 panic=1 rootfstype=ext4 rw console=ttyAMA0,115200" -hda ./outdisk -nographic`

**Comment by hpeyerl** [comment]
Thanks.  That works for me.  I had already tried adding CONFIG_FHANDLE=y  but I wasn't doing it with 4.1.18 plus it seems you have a few extra CONFIG_*'s in there ... shrug.

I had to modify the build script due to a problem with unzip on Ubuntu 14.04 with github .zip files .. I switched to .tar.gz instead of .zip instead of arguing with unzip...

https://github.com/hpeyerl/qemu-rpi-kernel/commit/9645bf437b804b778ae19785d0f471e2e7a4996d

(reference: https://github.com/raspberrypi/linux/issues/34 )
